spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/encryption/EncryptionController.java

```diff
@@ -169,6 +170,22 @@ private void checkEncryptorInstalled(String name, String profiles) {
 		}
 	}
 
+	private void checkDecryptionPossible(String name, String profiles) {
+		if (this.encryptor == null) {
```

Comment:
I would prefer we avoid duplicating the code from `chckEncryptorInstalled`.  Does the order of these checks matter?  Could we call `checkEncryptorInstalled` then check if we can actually decrypt it?

Reply:
Nope, the order doesn't matter. That's a remainder of trying to implement this functionality without touching `spring-security-rsa`. Will clean this up.